spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/handler/TaskJobLauncherCommandLineRunner.java

```diff
@@ -0,0 +1,195 @@
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionException;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersBuilder;
+import org.springframework.batch.core.JobParametersInvalidException;
+import org.springframework.batch.core.configuration.JobRegistry;
+import org.springframework.batch.core.converter.DefaultJobParametersConverter;
+import org.springframework.batch.core.converter.JobParametersConverter;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.launch.JobParametersNotFoundException;
+import org.springframework.batch.core.launch.NoSuchJobException;
+import org.springframework.batch.core.repository.JobExecutionAlreadyRunningException;
+import org.springframework.batch.core.repository.JobInstanceAlreadyCompleteException;
+import org.springframework.batch.core.repository.JobRestartException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.autoconfigure.batch.JobExecutionEvent;
+import org.springframework.cloud.task.listener.TaskException;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.core.Ordered;
+import org.springframework.util.PatternMatchUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link CommandLineRunner} to {@link JobLauncher launch} Spring Batch jobs. Runs all
+ * jobs in the surrounding context by default and throw an exception upon the
+ * first job that returns an {@link ExitStatus} of FAILED.
+ * Can also be used to launch a specific job by providing a jobName. The
+ * TaskJobLaunchercommandLineRunner takes the place of the
+ * {@link org.springframework.boot.autoconfigure.batch.JobLauncherCommandLineRunner}
+ * when it is in use.
+ *
+ * @author Glenn Renfro
+ * @since 2.0.0
+ */
+public class TaskJobLauncherCommandLineRunner implements CommandLineRunner, Ordered, ApplicationEventPublisherAware{
+	/**
+	 * The default order for the command line runner.
+	 */
+	public static final int DEFAULT_ORDER = 0;
+
+	private static final Log logger = LogFactory
+			.getLog(TaskJobLauncherCommandLineRunner.class);
+
+	private JobParametersConverter converter = new DefaultJobParametersConverter();
+
+	private JobLauncher jobLauncher;
+
+	private JobRegistry jobRegistry;
+
+	private JobExplorer jobExplorer;
+
+	private String jobNames;
+
+	private Collection<Job> jobs = Collections.emptySet();
```

Comment:
I may be missing something here.  The `FactoryBean` has no validation on the jobs being passed to this runner which means this could be null after the value has been set by the `FactoryBean`.  The setter for this is also marked as `@Autowired` (line 125) which, given the `FactoryBean` I think should be removed.  But since this field could be null as the code sits right now (unless I'm missing something), line 184 in this class could have an NPE because of this.

Reply:
Updated based on code review.